### DISCUSSION TITLE
Realm JavaScript v10.20.0-alpha.1

### DISCUSSION THREAD

**Discussion by kraenhansen** [realm/realm-js#3978] [General]
NOTE: DO NOT USE THIS RELEASE IN PRODUCTION!
NOTE: This is an early (alpha) release with Hermes/JSI support: We expect crashes and bugs.

### Enhancements
* Adding support for Hermes on Android.

### Fixed
* None.

### Compatibility
* MongoDB Realm Cloud.
* Realm Studio v11.0.0.
* APIs are backwards compatible with all previous releases of Realm JavaScript in the 10.5.x series.
* File format: generates Realms with format v22 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 or later for synced Realms).

### Internal
* Lint react-native templates and ensure they are checked by the CI.
* Small fix to Jenkins to publish Docker image for Raspberry Pi.

<hr /><em>This discussion was created from the release <a href='https://github.com/realm/realm-js/releases/tag/v10.20.0-alpha.1'>Realm JavaScript v10.20.0-alpha.1</a>.</em>

**Comment by 8BallBomBom** [comment]
Will test further but so far things seem to be working with RN 0.66.0-rc.3 along with Android 10 on a physical device 👍🏻

**Comment by batuhansahan** [comment]
So far there is no issue on IOS & Android. What is the roadmap, may i ask ? @kraenhansen

**Comment by harloware** [comment]
"realm": "10.20.0-alpha.1"
"react-native": "0.66.1"

I'm having an issue calling Realm Functions from the User class. This is on an existing project, and was working before the switch to 10.20.0-alpha.1.

LOG  [TypeError: Cannot read property 'findOrganizationsAssociatedByEmailAddress' of undefined]

I log the realmUser object prototype and this is the result:

["__Realm_internal", "constructor", "_customData", "accessToken", "apiKeys", "deviceId", "id", "identities", "isLoggedIn", "profile", "providerType", "refreshToken", "state", "_callFunction", "_linkCredentials", "_logOut", "_makeStreamingRequest", "_newWatchStream", "_pushDeregister", "_pushRegister", "_refreshCustomData", "_sessionForOnDiskPath", "toString", "toLocaleString", "valueOf", "hasOwnProperty", "isPrototypeOf", "propertyIsEnumerable", "__proto__", "__defineGetter__", "__defineSetter__", "__lookupGetter__", "__lookupSetter__"]

Edit: forgot to mention how I am invoking the realm function: 

const organizations = await realmUser.functions.findOrganizationsAssociatedByEmailAddress(emailAddress);

Update: I changed realm back to 10.91.1 and set :hermes_enabled => false, and then everything works as expected again. Logged the realmUser object prototype, and it looks as it should: 

["deviceId", "apiKeys", "accessToken", "providerType", "profile", "_customData", "refreshToken", "isLoggedIn", "state", "id", "identities", "_newWatchStream", "_pushRegister", "_linkCredentials", "_pushDeregister", "_refreshCustomData", "_logOut", "_makeStreamingRequest", "_callFunction", "_sessionForOnDiskPath", "linkCredentials", "logOut", "callFunction", "refreshCustomData", "mongoClient", "push", "functions", "auth", "customData", "_functionsOnService", "constructor", "toString", "toLocaleString", "valueOf", "hasOwnProperty", "propertyIsEnumerable", "isPrototypeOf", "__defineGetter__", "__defineSetter__", "__lookupGetter__", "__lookupSetter__", "__proto__"]

**Reply by 8BallBomBom** [reply]
Nice catch, not using the user functions on my end but the general database functions appear solid 👍🏻

**Reply by threefeetcat** [reply]
Have a similar issue here. 
Cannot logout - got TypeError: user.logOut is not a function
If I disable hermes & use the latest version of realm, everything works well.

**Reply by kraenhansen** [reply]
I've created https://github.com/realm/realm-js/issues/4117 to track this and @threefeetcat I've created https://github.com/realm/realm-js/issues/4118 to track your comment.

**Comment by somebody32** [comment]
In my case updating to that alpha generates an error when creating an object with a property of array of strings, ie:

```ts
realm.create(
          'Entity',
          {categories: ['personal']},
          UpdateMode.Modified
        );
```
gives:

`Error: Exception in HostFunction: Entity.categories must be of type 'string[]', got 'object' (personal), js engine: hermes`

but, interestingly, everything persists correctly.

But the good news is that now Realm works perfectly with Reanimated 2, reload works without a crash, so I believe that issue could be closed soon: https://github.com/software-mansion/react-native-reanimated/issues/1424

**Reply by whalemare** [reply]
In my case, hot-reload worked nicely on Androd, but `Reload` (with `R` button) cause crash of application

**Reply by kraenhansen** [reply]
I've created https://github.com/realm/realm-js/issues/4119 to track this.

**Comment by kraenhansen** [comment]
We've released a new version of Realm JS with support for the Hermes engine: [Realm JS v10.20.0-alpha.2](https://github.com/realm/realm-js/releases/tag/v10.20.0-alpha.2)

Please take it for a spin and provide any feedback or issues you might have using [the Hermes issue template](https://github.com/realm/realm-js/issues/new?assignees=&labels=hermes&template=hermes.yml).

**Reply by fronck** [reply]
@somebody32 Have you had a chance to look at whether this issue is still present in alpha.2?

**Reply by somebody32** [reply]
@fronck yes, unfortunately, it still happens

**Reply by fronck** [reply]
@somebody32 I haven't been able to reproduce the error.  Are you running the code in React Native?
Can you provide a full reproduction example?
Oh, and let's move the discussion on this issue to #4119.
